### PR TITLE
Add projections size to system.projections

### PR DESCRIPTION
**Pull request by jrdi**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add missing columns `total_rows`, `data_compressed_bytes`, and `data_uncompressed_bytes` to `system.projections`.
Part of https://github.com/ClickHouse/ClickHouse/pull/68901

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 8e9ec9b743989a9ec6d672661aaefc02899f737b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/70106/8e9ec9b743989a9ec6d672661aaefc02899f737b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/70106/8e9ec9b743989a9ec6d672661aaefc02899f737b/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Storages/System/StorageSystemProjections.cpp [hunk 1]
```diff
@@ -108,13 +112,15 @@ class ProjectionsSource : public ISource
...
                 const auto table = tables_it->table();
-                if (!table)
+                if (!table || !table->isMergeTree())
                     continue;
                 StorageMetadataPtr metadata_snapshot = table->getInMemoryMetadataPtr();
                 if (!metadata_snapshot)
                     continue;
                 const auto & projections = metadata_snapshot->getProjections();
 
+                auto parts = dynamic_cast<const MergeTreeData &>(*table).getDataPartsVectorForInternalUsage({MergeTreeData::DataPartState::Active}, nullptr);
```
Comment (nikitamikhaylov) [inline src/Storages/System/StorageSystemProjections.cpp:122]: Why not to use `typeid` cast? You've checked that the table is a MergeTree in the code above.

**Comment by jrdi** [comment]
@alexey-milovidov, I was rebasing from master and didn't notice that you pushed a commit. I have overwritten it, can you push it again?

**Comment by nikitamikhaylov** [comment]
@jrdi This was the commit: https://github.com/ClickHouse/ClickHouse/commit/8e9ec9b743989a9ec6d672661aaefc02899f737b. You can push it yourself.

**Comment by jrdi** [comment]
> AST fuzzer (ubsan) — Logical error: 'Too many keys are used for CubeTransform.'

Seems unrelated to this PR.

**Comment by alexey-milovidov** [comment]
> Seems unrelated to this PR.

Yes, but we have to fix them first...

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/pull/70770

**Comment by jrdi** [comment]
> Yes, but we have to fix them first...

Thank you for fixing the remaining tests. We are green now 🎉

**Review by alexey-milovidov — APPROVED** [review]

**Comment by amosbird** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/70853/d6dace0a595cc427beaf118289e7871e305fce6d/stress_test__tsan_/fatal_messages.txt

Seems related.

**Comment by alexey-milovidov** [comment]
@amosbird, thanks. Reverted.

**Pull request merged**